test/unit/compiler/passes/report-infinite-repetition.spec.js

```diff
@@ -258,6 +258,8 @@ describe("compiler pass |reportInfiniteRepetition|", () => {
 
     expect(pass).to.reportError("start = (!{ })*");
 
+    expect(pass).to.reportError("start = 'a'|..,([ \t]*)+'.'|");
```

Comment:
I think it would better to use existing primitive that do not consume any input, and also add a positive test:
```suggestion
    expect(pass).to.reportError("start = 'a'|.., ('')*|");
    expect(pass).to.not.reportError("start = 'a'|.., ('b')*|");
```